evm/src/cpu/kernel/interpreter.rs

```diff
@@ -2,50 +2,32 @@ use anyhow::{anyhow, bail};
 use ethereum_types::{BigEndianHash, U256, U512};
 use keccak_hash::keccak;
 
+use crate::generation::memory::MemoryContextState;
+use crate::memory::segments::Segment;
+
 /// Halt interpreter execution whenever a jump to this offset is done.
 const HALT_OFFSET: usize = 0xdeadbeef;
 
-#[derive(Debug, Default)]
-pub(crate) struct EvmMemory {
-    memory: Vec<u8>,
+#[derive(Debug)]
+pub(crate) struct ContextMemory {
```

Comment:
nit: I feel like "context" would be better in the field name than the struct name, since the fact that the outer Vec is for the context dimension seems like an internal thing; in the API context is just one of three.

Reply:
Good point, I've renamed to
```rust
struct InterpreterMemory {
    context_memory: Vec<MemoryContextState>,
}
```